simplex/src/main/java/io/em2m/simplex/std/dates.kt

```diff
@@ -0,0 +1,107 @@
+package io.em2m.simplex.std
+
+import io.em2m.simplex.model.*
+import io.em2m.simplex.parser.DateMathParser
+import io.em2m.utils.coerce
+import io.em2m.utils.coerceNonNull
+import org.joda.time.format.DateTimeFormat
+import java.util.*
+
+class FormatDatePipe : PipeTransform {
+
+    private var dateFormat: String = ""
+
+    override fun args(args: List<String>) {
+        if (args.isNotEmpty()) {
+            dateFormat = args[0]
+        }
+    }
+
+    override fun transform(value: Any?, context: ExprContext): Any? {
+        val dateInput : Date? = value?.coerce()
+         return if (dateInput != null) {
+             val pattern = DateTimeFormat.forPattern(dateFormat)
+             pattern.print(dateInput.time)
+         } else value
+    }
+}
+
+class DateMathPipe : PipeTransform {
+
+    private val dateMathParser = DateMathParser()
+    private var dateMath: String = ""
```

Comment:
This should be null by default and we should check for null below.  It's actually an error if no value is provided.  We need to decide how to handle errors.